controlpanel/frontend/views/auth.py

```diff
@@ -62,3 +63,17 @@ def update_user(self, token):
         email = token["userinfo"]["email"]
         self.request.user.justice_email = email
         self.request.user.save()
+
+
+class EntraOIDCAuthenticationCallbackView(OIDCAuthenticationCallbackView):
```

Comment:
This does the job, and works. However the project already overrides the callback view to us this https://github.com/ministryofjustice/analytics-platform-control-panel/blob/main/controlpanel/oidc.py#L67 

Then there is a setting that points to that class in the oidc package https://github.com/ministryofjustice/analytics-platform-control-panel/blob/main/controlpanel/settings/common.py#L164

So we dont need to override the URL, and also dont want to lose that additional functionality that has already been added. But we can still use your logic for the redirect on that view - if you look in the source code, the `OIDCAuthenticationCallbackView` has a `success_url` method. So I would suggest we add that method to the `StateMismatchHandler` view (could rename the view if you wanted), to check for the email and return the index url if we havent captured it, or call `super()` if we already have it.

Reply:
Would also be good to add a unit test - should be easier if all the logic is in the `success_url` method